public/app/features/provisioning/utils.ts

```diff
@@ -0,0 +1,10 @@
+import { RepositoryViewList } from './api';
+
+export function checkSyncSettings(settings: RepositoryViewList | undefined): [boolean, boolean] {
+  if (!settings?.items?.length) {
+    return [false, false];
+  }
+  const instanceConnected = settings.items.some((item) => item.target === 'instance');
+  const folderConnected = settings.items.some((item) => item.target === 'folder');
```

Comment:
Aren't these two mutually exclusive, e.g., it can be either instance or folder connected?

Reply:
There is also the third situation in which nothing is connected. Can you guide on what the nicest syntax for that will be?

Reply:
I'd just use separate `isInstance` and `isFolder` functions, but having an empty options complicates that. I guess the current approach is ok then.